src/core/patch.rkt

```diff
@@ -104,11 +86,10 @@
   ; apply changelists
   (define rewritten
     (reap [sow]
-          (for ([changelists changelistss] [altn altns] [repr reprs])
+          (for ([changelists changelistss] [altn altns])
             (for ([cl changelists])
               (match-define (list subexpr input) cl)
-              (when (impl-well-typed? subexpr repr)
```

Comment:
Do you remember why this was needed in the first place? I recall it being something super subtle.